src/main/webapp/app/shared/source/source.service.ts

```diff
@@ -81,4 +67,15 @@ export class SourceService {
         };
         return this.http.get(this.resourceUrl, options);
     }
+
+    findAvailable(req?: any): Observable<Response> {
+        const params: URLSearchParams = new URLSearchParams();
+        if (req) {
+            params.set('assigned', req.assigned);
+        }
+        const options = {
+            search: params
+        };
+        return this.http.get(`${this.projectResourceUrl}/${req.projectId}/sources` , options);
```

Comment:
req can be null

Reply:
After the merge of this PR i'm rebasing and pushing a branch and PR for #87, where this will be fixed.